app/src/main/java/ch/epfl/sdp/appart/Database.java

```diff
@@ -1,17 +1,16 @@
 package ch.epfl.sdp.appart;
 
-import com.google.android.gms.tasks.OnCompleteListener;
-import com.google.firebase.firestore.DocumentReference;
-import com.google.firebase.firestore.QuerySnapshot;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
```

Comment:
Should I use completable futures instead of LoginCallback for the auth service ?

Reply:
Do as you wish, I think in the data base it makes more sens, I don't know for the login. You see that back then we were licking some Firebase classes and that wasn't good. Plus, the callback implied that the querying class need to know some implementation details of the database which I found shady.